core/sail/api/src/main/java/org/eclipse/rdf4j/sail/helpers/AbstractSailConnection.java

```diff
@@ -459,6 +459,10 @@ public final void rollback()
 						txnPrepared = false;
 					}
 				}
+				else {
+					logger.warn("Cannot rollback transaction on connection because transaction is not active",
+							new Throwable());
```

Comment:
The current system property that is used for AbstractSailConnection.debugEnabled is "org.eclipse.rdf4j.repository.debug", as setup in AbstractSail:

https://github.com/eclipse/rdf4j/blob/a283e758fb79e7b62e94b669547e6e64c64106cf/core/sail/api/src/main/java/org/eclipse/rdf4j/sail/helpers/AbstractSail.java#L66

Agree that we should migrate the lock debugging property, "info.aduna.concurrent.locks.trackLocks", but probably useful to grandfather the existing property name for a few versions.